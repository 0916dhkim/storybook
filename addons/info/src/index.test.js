/* global document */

import React from 'react';
import ReactDOM from 'react-dom';
import AddonInfo, { withInfo } from './';

describe('addon Info', () => {
  const story = context =>
    <div>
      It's a {context.story} story
    </div>;
  const api = {
    add: (name, fn) => fn({ kind: 'addon_info', story: 'jest_test' }),
  };
  it('should render <Info />', () => {
    const Info = withInfo('Test Info')(story);
    ReactDOM.render(<Info />, document.createElement('div'));
  });
  it('should show deprecation warning', () => {
    const addWithInfo = AddonInfo.addWithInfo.bind(api);
    addWithInfo('jest', 'test info', story);
  });
});
