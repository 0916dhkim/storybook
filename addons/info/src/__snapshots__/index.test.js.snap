// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addon Info should render <Info /> and external markdown 1`] = `
<Component>
  <Story
    PropTable={[Function]}
    components={
      Object {
        "a": [Function],
        "code": [Function],
        "h1": [Function],
        "h2": [Function],
        "h3": [Function],
        "h4": [Function],
        "h5": [Function],
        "h6": [Function],
        "li": [Function],
        "p": [Function],
        "ul": [Function],
      }
    }
    context={Object {}}
    excludedPropTypes={Array []}
    info=<h1>
  HTML Mock
</h1>
    maxPropArrayLength={3}
    maxPropObjectKeys={3}
    maxPropStringLength={50}
    maxPropsIntoLine={3}
    propTableCompare={[Function]}
    propTables={Array []}
    propTablesExclude={Array []}
    showHeader={true}
    showInline={false}
    showSource={true}
    styles={[Function]}
  >
    <div
      style={
        Object {
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      <div>
        It's a 
         story:
        <TestComponent
          array={
            Array [
              1,
              2,
              3,
            ]
          }
          bool={true}
          func={[Function]}
          number={7}
          obj={
            Object {
              "a": "a",
              "b": "b",
            }
          }
          string="seven"
        >
          <div>
            <h1>
              x =&gt; x + 1
            </h1>
            <h2>
              [object Object]
            </h2>
            <h3>
              1,2,3
            </h3>
            <h4>
              7
            </h4>
            <h5>
              seven
            </h5>
            <h6>
              true
            </h6>
            <p>
              undefined
            </p>
            <a
              href="#"
            >
              test
            </a>
            <code>
              storiesOf
            </code>
            <ul>
              <li>
                1
              </li>
              <li>
                2
              </li>
            </ul>
          </div>
        </TestComponent>
      </div>
    </div>
    <button
      className="info__show-button"
      onClick={[Function]}
      style={
        Object {
          "background": "#027ac5",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": 12,
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Show Info
    </button>
  </Story>
</Component>
`;

exports[`addon Info should render <Info /> and markdown 1`] = `
<Component>
  <Story
    PropTable={[Function]}
    components={
      Object {
        "a": [Function],
        "code": [Function],
        "h1": [Function],
        "h2": [Function],
        "h3": [Function],
        "h4": [Function],
        "h5": [Function],
        "h6": [Function],
        "li": [Function],
        "p": [Function],
        "ul": [Function],
      }
    }
    context={Object {}}
    excludedPropTypes={Array []}
    info="# Test story
## with markdown info
containing **bold**, *cursive* text, \`code\` and [a link](https://github.com)"
    maxPropArrayLength={3}
    maxPropObjectKeys={3}
    maxPropStringLength={50}
    maxPropsIntoLine={3}
    propTableCompare={[Function]}
    propTables={Array []}
    propTablesExclude={Array []}
    showHeader={true}
    showInline={false}
    showSource={true}
    styles={[Function]}
  >
    <div
      style={
        Object {
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      <div>
        It's a 
         story:
        <TestComponent
          array={
            Array [
              1,
              2,
              3,
            ]
          }
          bool={true}
          func={[Function]}
          number={7}
          obj={
            Object {
              "a": "a",
              "b": "b",
            }
          }
          string="seven"
        >
          <div>
            <h1>
              x =&gt; x + 1
            </h1>
            <h2>
              [object Object]
            </h2>
            <h3>
              1,2,3
            </h3>
            <h4>
              7
            </h4>
            <h5>
              seven
            </h5>
            <h6>
              true
            </h6>
            <p>
              undefined
            </p>
            <a
              href="#"
            >
              test
            </a>
            <code>
              storiesOf
            </code>
            <ul>
              <li>
                1
              </li>
              <li>
                2
              </li>
            </ul>
          </div>
        </TestComponent>
      </div>
    </div>
    <button
      className="info__show-button"
      onClick={[Function]}
      style={
        Object {
          "background": "#027ac5",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": 12,
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      Show Info
    </button>
  </Story>
</Component>
`;

exports[`addon Info should render <Info /> for memoized component 1`] = `
<div style="position: relative; z-index: 0;">
  <div>
    It's a  story:
    <div>
      <h1>
        x =&gt; x + 1
      </h1>
      <h2>
        [object Object]
      </h2>
      <h3>
        1,2,3
      </h3>
      <h4>
        7
      </h4>
      <h5>
        seven
      </h5>
      <h6>
        true
      </h6>
      <p>
        undefined
      </p>
      <a href="#">
        test
      </a>
      <code>
        storiesOf
      </code>
      <ul>
        <li>
          1
        </li>
        <li>
          2
        </li>
      </ul>
    </div>
  </div>
</div>
<button type="button"
        style="font-family: sans-serif; font-size: 12px; display: block; position: fixed; background: rgb(2, 122, 197); color: rgb(255, 255, 255); padding: 5px 15px; cursor: pointer; top: 0px; right: 0px; border-radius: 0 0 0 5px;"
        class="info__show-button"
>
  Show Info
</button>
`;

exports[`addon Info should render component description if story kind matches component 1`] = `
<div>
  <div style="font-family: Helvetica Neue, Helvetica, Segoe UI, Arial, freesans, sans-serif; color: black; font-weight: 300; line-height: 1.45; font-size: 15px; padding: 20px 40px 40px; border-radius: 2px; background-color: rgb(255, 255, 255);">
    <div style="border-bottom: 1px solid #eee; padding-top: 10px; margin-bottom: 10px;">
      <h1 style="margin: 0px; padding: 0px; font-size: 35px;">
        TestComponent
      </h1>
      <h2 style="margin: 0px 0px 10px 0px; padding: 0px; font-weight: 400; font-size: 22px;">
        Basic test
      </h2>
    </div>
  </div>
</div>
<div id="story-root">
  <div>
    It's a Basic test story:
    <div>
      <h1>
        x =&gt; x + 1
      </h1>
      <h2>
        [object Object]
      </h2>
      <h3>
        1,2,3
      </h3>
      <h4>
        7
      </h4>
      <h5>
        seven
      </h5>
      <h6>
        true
      </h6>
      <p>
        undefined
      </p>
      <a href="#">
        test
      </a>
      <code>
        storiesOf
      </code>
      <ul>
        <li>
          1
        </li>
        <li>
          2
        </li>
      </ul>
    </div>
  </div>
</div>
<div>
  <div style="font-family: Helvetica Neue, Helvetica, Segoe UI, Arial, freesans, sans-serif; color: black; font-weight: 300; line-height: 1.45; font-size: 15px; padding: 20px 40px 40px; border-radius: 2px; background-color: rgb(255, 255, 255);">
    <h1 id="awesome-test-component-description"
        style="border-bottom: 1px solid #eee; font-weight: 600; margin: 0px; padding: 0px; font-size: 40px;"
    >
      Awesome test component description
    </h1>
    <h2 id="awesome-test-component-description-with-markdown-support"
        style="font-weight: 600; margin: 0px; padding: 0px; font-size: 30px;"
    >
      with markdown support
    </h2>
    <p>
      <strong>
        bold
      </strong>
      <em>
        cursive
      </em>
    </p>
    <div class="css-1fhpnuv">
      <div class="css-1si67pu os-host os-theme-dark os-host-resize-disabled os-host-scrollbar-horizontal-hidden os-host-scrollbar-vertical-hidden">
        <div class="os-resize-observer-host">
          <div class="os-resize-observer"
               style="right: 0px;"
          >
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-size-auto-observer"
             style="float: left;"
        >
          <div class="os-resize-observer">
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-content-glue"
             style="margin: 0px 0px 0px 0px; width: -1px; height: -1px;"
        >
        </div>
        <div class="os-padding">
          <div class="os-viewport os-viewport-native-scrollbars-overlaid"
               style
          >
            <div class="os-content-arrange">
            </div>
            <div class="os-content"
                 style="padding: 0px 0px 0px 0px; height: 100%; width: 100%;"
            >
              <pre class="hljs css-4zr3vl">
                <code class="css-1maezg8">
                  a
                  <span class="token punctuation">
                    ;
                  </span>
                </code>
              </pre>
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-horizontal os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0px, 0);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-vertical os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0, 0px);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar-corner">
        </div>
      </div>
      <div class="css-11xgcgt">
        <button class="css-1imo1gr">
          Copy
        </button>
      </div>
    </div>
    <h1 style="margin: 20px 0px 0px 0px; padding: 0px 0px 5px 0px; font-size: 25px; border-bottom: 1px solid #EEE;">
      Story Source
    </h1>
    <div class="css-1fhpnuv">
      <div class="css-1si67pu os-host os-theme-dark os-host-resize-disabled os-host-scrollbar-horizontal-hidden os-host-scrollbar-vertical-hidden">
        <div class="os-resize-observer-host">
          <div class="os-resize-observer"
               style="right: 0px;"
          >
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-size-auto-observer"
             style="float: left;"
        >
          <div class="os-resize-observer">
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-content-glue"
             style="margin: 0px 0px 0px 0px; width: -1px; height: -1px;"
        >
        </div>
        <div class="os-padding">
          <div class="os-viewport os-viewport-native-scrollbars-overlaid"
               style
          >
            <div class="os-content-arrange">
            </div>
            <div class="os-content"
                 style="padding: 0px 0px 0px 0px; height: 100%; width: 100%;"
            >
              <pre class="hljs css-4zr3vl">
                <code class="css-1maezg8">
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;
                      </span>
                      div
                    </span>
                    <span class="token punctuation">
                      &gt;
                    </span>
                  </span>
                  <span class="token plain-text">
                    It's a Basic test story:
                  </span>
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;
                      </span>
                      <span class="token class-name">
                        TestComponent
                      </span>
                    </span>
                    <span class="token attr-name">
                      array
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        [
                      </span>
                      <span class="token number">
                        1
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      <span class="token number">
                        2
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      <span class="token number">
                        3
                      </span>
                      <span class="token punctuation">
                        ]
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      bool
                    </span>
                    <span class="token attr-name">
                      func
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token keyword">
                        function
                      </span>
                      <span class="token function">
                        noRefCheck
                      </span>
                      <span class="token punctuation">
                        (
                      </span>
                      <span class="token punctuation">
                        )
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      number
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token number">
                        7
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      obj
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      a
                      <span class="token punctuation">
                        :
                      </span>
                      <span class="token string">
                        'a'
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      b
                      <span class="token punctuation">
                        :
                      </span>
                      <span class="token string">
                        'b'
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      string
                    </span>
                    <span class="token attr-value">
                      <span class="token punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        "
                      </span>
                      seven
                      <span class="token punctuation">
                        "
                      </span>
                    </span>
                    <span class="token punctuation">
                      /&gt;
                    </span>
                  </span>
                  <span class="token plain-text">
                  </span>
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;/
                      </span>
                      div
                    </span>
                    <span class="token punctuation">
                      &gt;
                    </span>
                  </span>
                </code>
              </pre>
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-horizontal os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0px, 0);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-vertical os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0, 0px);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar-corner">
        </div>
      </div>
      <div class="css-11xgcgt">
        <button class="css-1imo1gr">
          Copy
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`addon Info should render component description if story name matches component 1`] = `
<div>
  <div style="font-family: Helvetica Neue, Helvetica, Segoe UI, Arial, freesans, sans-serif; color: black; font-weight: 300; line-height: 1.45; font-size: 15px; padding: 20px 40px 40px; border-radius: 2px; background-color: rgb(255, 255, 255);">
    <div style="border-bottom: 1px solid #eee; padding-top: 10px; margin-bottom: 10px;">
      <h1 style="margin: 0px; padding: 0px; font-size: 35px;">
        Test Components
      </h1>
      <h2 style="margin: 0px 0px 10px 0px; padding: 0px; font-weight: 400; font-size: 22px;">
        TestComponent
      </h2>
    </div>
  </div>
</div>
<div id="story-root">
  <div>
    It's a TestComponent story:
    <div>
      <h1>
        x =&gt; x + 1
      </h1>
      <h2>
        [object Object]
      </h2>
      <h3>
        1,2,3
      </h3>
      <h4>
        7
      </h4>
      <h5>
        seven
      </h5>
      <h6>
        true
      </h6>
      <p>
        undefined
      </p>
      <a href="#">
        test
      </a>
      <code>
        storiesOf
      </code>
      <ul>
        <li>
          1
        </li>
        <li>
          2
        </li>
      </ul>
    </div>
  </div>
</div>
<div>
  <div style="font-family: Helvetica Neue, Helvetica, Segoe UI, Arial, freesans, sans-serif; color: black; font-weight: 300; line-height: 1.45; font-size: 15px; padding: 20px 40px 40px; border-radius: 2px; background-color: rgb(255, 255, 255);">
    <h1 id="awesome-test-component-description"
        style="border-bottom: 1px solid #eee; font-weight: 600; margin: 0px; padding: 0px; font-size: 40px;"
    >
      Awesome test component description
    </h1>
    <h2 id="awesome-test-component-description-with-markdown-support"
        style="font-weight: 600; margin: 0px; padding: 0px; font-size: 30px;"
    >
      with markdown support
    </h2>
    <p>
      <strong>
        bold
      </strong>
      <em>
        cursive
      </em>
    </p>
    <div class="css-1fhpnuv">
      <div class="css-1si67pu os-host os-theme-dark os-host-resize-disabled os-host-scrollbar-horizontal-hidden os-host-scrollbar-vertical-hidden">
        <div class="os-resize-observer-host">
          <div class="os-resize-observer"
               style="right: 0px;"
          >
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-size-auto-observer"
             style="float: left;"
        >
          <div class="os-resize-observer">
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-content-glue"
             style="margin: 0px 0px 0px 0px; width: -1px; height: -1px;"
        >
        </div>
        <div class="os-padding">
          <div class="os-viewport os-viewport-native-scrollbars-overlaid"
               style
          >
            <div class="os-content-arrange">
            </div>
            <div class="os-content"
                 style="padding: 0px 0px 0px 0px; height: 100%; width: 100%;"
            >
              <pre class="hljs css-4zr3vl">
                <code class="css-1maezg8">
                  a
                  <span class="token punctuation">
                    ;
                  </span>
                </code>
              </pre>
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-horizontal os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0px, 0);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-vertical os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0, 0px);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar-corner">
        </div>
      </div>
      <div class="css-11xgcgt">
        <button class="css-1imo1gr">
          Copy
        </button>
      </div>
    </div>
    <h1 style="margin: 20px 0px 0px 0px; padding: 0px 0px 5px 0px; font-size: 25px; border-bottom: 1px solid #EEE;">
      Story Source
    </h1>
    <div class="css-1fhpnuv">
      <div class="css-1si67pu os-host os-theme-dark os-host-resize-disabled os-host-scrollbar-horizontal-hidden os-host-scrollbar-vertical-hidden">
        <div class="os-resize-observer-host">
          <div class="os-resize-observer"
               style="right: 0px;"
          >
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-size-auto-observer"
             style="float: left;"
        >
          <div class="os-resize-observer">
            <div class="os-resize-observer-item"
                 dir="ltr"
                 style="top: -2px; right: 0px; bottom: 0px; left: -2px;"
            >
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 3333333px; height: 3333333px;"
                >
                </div>
              </div>
              <div class="os-resize-observer-item">
                <div class="os-resize-observer-item-final"
                     style="width: 200%; height: 200%"
                >
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="os-content-glue"
             style="margin: 0px 0px 0px 0px; width: -1px; height: -1px;"
        >
        </div>
        <div class="os-padding">
          <div class="os-viewport os-viewport-native-scrollbars-overlaid"
               style
          >
            <div class="os-content-arrange">
            </div>
            <div class="os-content"
                 style="padding: 0px 0px 0px 0px; height: 100%; width: 100%;"
            >
              <pre class="hljs css-4zr3vl">
                <code class="css-1maezg8">
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;
                      </span>
                      div
                    </span>
                    <span class="token punctuation">
                      &gt;
                    </span>
                  </span>
                  <span class="token plain-text">
                    It's a TestComponent story:
                  </span>
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;
                      </span>
                      <span class="token class-name">
                        TestComponent
                      </span>
                    </span>
                    <span class="token attr-name">
                      array
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        [
                      </span>
                      <span class="token number">
                        1
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      <span class="token number">
                        2
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      <span class="token number">
                        3
                      </span>
                      <span class="token punctuation">
                        ]
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      bool
                    </span>
                    <span class="token attr-name">
                      func
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token keyword">
                        function
                      </span>
                      <span class="token function">
                        noRefCheck
                      </span>
                      <span class="token punctuation">
                        (
                      </span>
                      <span class="token punctuation">
                        )
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      number
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token number">
                        7
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      obj
                    </span>
                    <span class="token script language-javascript">
                      <span class="token script-punctuation punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      <span class="token punctuation">
                        {
                      </span>
                      a
                      <span class="token punctuation">
                        :
                      </span>
                      <span class="token string">
                        'a'
                      </span>
                      <span class="token punctuation">
                        ,
                      </span>
                      b
                      <span class="token punctuation">
                        :
                      </span>
                      <span class="token string">
                        'b'
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                      <span class="token punctuation">
                        }
                      </span>
                    </span>
                    <span class="token attr-name">
                      string
                    </span>
                    <span class="token attr-value">
                      <span class="token punctuation">
                        =
                      </span>
                      <span class="token punctuation">
                        "
                      </span>
                      seven
                      <span class="token punctuation">
                        "
                      </span>
                    </span>
                    <span class="token punctuation">
                      /&gt;
                    </span>
                  </span>
                  <span class="token plain-text">
                  </span>
                  <span class="token tag">
                    <span class="token tag">
                      <span class="token punctuation">
                        &lt;/
                      </span>
                      div
                    </span>
                    <span class="token punctuation">
                      &gt;
                    </span>
                  </span>
                </code>
              </pre>
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-horizontal os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0px, 0);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar os-scrollbar-vertical os-scrollbar-unusable os-scrollbar-auto-hidden">
          <div class="os-scrollbar-track os-scrollbar-track-off">
            <div class="os-scrollbar-handle"
                 style="transform: translate(0, 0px);"
            >
            </div>
          </div>
        </div>
        <div class="os-scrollbar-corner">
        </div>
      </div>
      <div class="css-11xgcgt">
        <button class="css-1imo1gr">
          Copy
        </button>
      </div>
    </div>
  </div>
</div>
`;
